src/nwm_routing/src/nwm_routing/preprocess.py

```diff
@@ -292,35 +296,37 @@ def nwm_initial_warmstate_preprocess(
             waterbodies_df, waterbodies_initial_states_df, on="lake_id"
         )
 
-        
         LOG.debug("waterbody initial states complete in %s seconds." % (time.time() - start_time))
         start_time = time.time()
 
     # STEP 4: Handle Channel Initial States, set T0, and initialize LastObs
-
     start_time = time.time()
-
     LOG.info("setting channel initial states ...")
 
-    q0 = nnu.build_channel_initial_state(restart_parameters, segment_index)
-
     # STEP 4a: Set Channel States and T0
-    if restart_parameters.get("wrf_hydro_channel_restart_file", None):
-        channel_initial_states_file = restart_parameters[
-            "wrf_hydro_channel_restart_file"
-        ]
-        t0_str = nhd_io.get_param_str(channel_initial_states_file, "Restart_Time")
+    if restart_parameters.get("lite_channel_restart_file", None):
+        
+        q0, t0 = nhd_io.read_lite_restart(
+            restart_parameters['lite_channel_restart_file']
+        )
```

Comment:
If a lite channel restart file is provided, it is read-in and used with preference over `HYDRO_RST` or `.csv`. Here, the `t0` variable is returned from `nhd_io.read_lite_restart` and subsequently passed back to `__main__.py`